appveyor.yml

```diff
@@ -1,26 +1,67 @@
-  - cinst -y OpenSSL.Light
-  - SET PATH=C:\Program Files\OpenSSL;%PATH%
-  - sc config wuauserv start= auto
-  - net start wuauserv
-  - cinst -y php
-  - cd c:\tools\php71
-  - copy php.ini-production php.ini
-  - echo date.timezone="UTC" >> php.ini
-  - echo extension_dir=ext >> php.ini
-  - echo extension=php_openssl.dll >> php.ini
-  - echo extension=php_mbstring.dll >> php.ini
-  - cd C:\projects\better-reflection
-  - SET PATH=C:\tools\php71;%PATH%
-  - php -r "readfile('http://getcomposer.org/installer');" | php
-  - php composer.phar install --prefer-source --no-interaction
+  # Download requested PHP version
+  - ps: If ((Test-Path $Env:php_archive_directory) -eq $False) { New-Item -Path $Env:php_archive_directory -ItemType 'directory' }
+  - ps: $requested_php_version = (((choco search php --exact --all-versions -r | Select-String -pattern $Env:php_version) -replace '[php|]', '') | %{ New-Object System.Version $_ } | Sort-Object | Select-Object -Last 1).ToString()
+  - ps: $php_version_url = 'http://windows.php.net/downloads/releases/php-' + $requested_php_version + '-nts-Win32-VC14-' + $Env:platform + '.zip'
+  - ps: $php_version_file = $Env:php_archive_directory + '\php-' + $requested_php_version + '.zip'
+  - ps: If ((Test-Path $php_version_file) -eq $False) { appveyor-retry appveyor DownloadFile $php_version_url -FileName $php_version_file }
 
+  # Prepare PHP
+  - ps: New-Item -Path $Env:php_directory -ItemType 'directory'
+  - ps: cd $Env:php_directory
+  - ps: 7z x $php_version_file
+  - ps: Copy-Item php.ini-production -Destination php.ini
+  - ps: Add-Content -Path php.ini -Value 'date.timezone="UTC"'
+  - ps: Add-Content -Path php.ini -Value 'extension_dir=ext'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_curl.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_intl.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_openssl.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_mbstring.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_mysqli.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_pdo_mysql.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_soap.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_sqlite3.dll'
+  - ps: Add-Content -Path php.ini -Value 'extension=php_xsl.dll'
+  - ps: php --version
+
+  # Prepare composer
+  - ps: If ((Test-Path $Env:composer_directory) -eq $False) { New-Item -Path $Env:composer_directory -ItemType 'directory' }
+  - ps: If ((Test-Path $Env:composer_executable) -eq $False) { appveyor-retry appveyor DownloadFile https://getcomposer.org/composer.phar -FileName $Env:composer_executable }
+  - ps: Set-Content -Path ($Env:composer_directory + '\composer.bat') -Value ('@php ' + $Env:composer_executable + ' %*')
+  - composer self-update
+
+  # Install dependencies
+  - ps: cd $Env:project_directory
+  - IF %dependencies%==lowest composer update --prefer-lowest --prefer-dist --no-interaction --no-progress
```

Comment:
I use the same `--prefer-*` parameters that are in travis.yml: https://github.com/Roave/BetterReflection/blob/master/.travis.yml#L12